plugins/service-monitor-activemq/src/main/java/com/thinkbiganalytics/service/activemq/ActivemqServiceStatusCheck.java

```diff
@@ -0,0 +1,132 @@
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+
+import com.thinkbiganalytics.servicemonitor.check.ServiceStatusCheck;
+import com.thinkbiganalytics.servicemonitor.model.DefaultServiceAlert;
+import com.thinkbiganalytics.servicemonitor.model.DefaultServiceComponent;
+import com.thinkbiganalytics.servicemonitor.model.DefaultServiceStatusResponse;
+import com.thinkbiganalytics.servicemonitor.model.ServiceAlert;
+import com.thinkbiganalytics.servicemonitor.model.ServiceComponent;
+import com.thinkbiganalytics.servicemonitor.model.ServiceStatusResponse;
+
+import org.apache.activemq.ActiveMQConnection;
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+public class ActivemqServiceStatusCheck implements ServiceStatusCheck{
+
+    private static final Logger log = LoggerFactory.getLogger(ActivemqServiceStatusCheck.class);
+
+    @Value("${jms.activemq.broker.url:#{null}}")
+    private String activemqBrokerUrl;
+
+
+    static final  String SERVICE_NAME = "Activemq"; 
+
+    @Autowired
+    private Environment env;
+
+    @Override
+    public ServiceStatusResponse healthCheck() {
+
+
+        String serviceName = SERVICE_NAME;
+
+        return new DefaultServiceStatusResponse(serviceName, Arrays.asList(activemqStatus()));
+    }
+
+    /**
+     * Check if Activemq is running
+     * 
+     * @return Status of Activemq
+     */
+
+    private ServiceComponent activemqStatus() {
+
+        String componentName = "Activemq";
+        String serviceName = SERVICE_NAME;
+        ServiceComponent component = null;
+
+
+        /**
+         * Prepare Alert Message
+         */
+        ServiceAlert alert = null;
+        alert = new DefaultServiceAlert();
+        alert.setLabel(componentName);
+        alert.setServiceName(serviceName);
+        alert.setComponentName(componentName);
+
+
+        String finalServiceMessage = ""; 
+
+        try {
+
+
+            if ( StringUtils.isNotBlank(activemqBrokerUrl) )
+            {
+
+                ActiveMQConnection connection = null;
+
+                String activemqConnectionUri = activemqBrokerUrl;
+
+
+                ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory(activemqConnectionUri);
+
+                /**
+                 * Exception is thrown if connection failed.
+                 */
+                connection = (ActiveMQConnection)factory.createConnection();
```

Comment:
The team discussed the connection create mechanism here.  We believe this is too heavy weight, as you will create a new stateful ActiveMQConnection on every status check.  In the code base there should be a Refeshable connection class you could use.   I don't have a pointer for it now.  If you can't find it let me know.